### PR TITLE
Better OpenAPI Enums

### PR DESCRIPTION
**Pull request by 1lutz**
Based on https://github.com/geo-engine/geoengine/pull/953

### CODE REVIEW

File: .generation/generate.py [hunk 1]
```diff
@@ -174,6 +174,7 @@ def generate_python_code(*, package_name: str, package_version: str, package_url
         ],
         check=True,
     )
+    shutil.rmtree(Path("python") / "docs")
```
Comment (michaelmattig) [inline .generation/generate.py:177]: why did you add this?

File: .generation/generate.py [hunk 2]
```diff
@@ -208,6 +209,11 @@ def generate_typescript_code(*, npm_name: str, npm_version: str, repository_url:
         ],
         check=True,
     )
+    with open(Path("typescript") / ".gitignore", 'w', encoding='utf-8') as f:
```
Comment (michaelmattig) [inline .generation/generate.py:212]: why did you add this?

File: python/.openapi-generator/FILES [hunk 3]
```diff
@@ -1,4 +1,236 @@
+.github/workflows/python.yml
+.gitignore
+.gitlab-ci.yml
+.openapi-generator-ignore
+.travis.yml
 README.md
+docs/AddCollection200Response.md
```
Comment (michaelmattig) [inline python/.openapi-generator/FILES:7]: are this files here correct? you delete the docs in the generate.py right?

File: typescript/.openapi-generator-ignore [hunk 4]
```diff
@@ -1,22 +1,23 @@
 # OpenAPI Generator Ignore
-
-# Used by openapi-generator https://github.com/openapitools/openapi-generator
+# Generated by openapi-generator https://github.com/openapitools/openapi-generator
```
Comment (michaelmattig) [inline typescript/.openapi-generator-ignore:2]: why did this change? Is it from the new version? so maybe roll it back as well

File: .generation/generate.py (hunk 1 above)
Comment (1lutz) [reply .generation/generate.py:177]: Because at the moment in the `main`branch there are no docs, so I thought they are not wanted in the package. But the generator outputs docs so I delete them.

File: typescript/.openapi-generator-ignore (hunk 4 above)
Comment (1lutz) [reply typescript/.openapi-generator-ignore:2]: I do not know. I completely deleted the client folders, built the old generator image and regenerated the files.

File: .generation/generate.py (hunk 2 above)
Comment (1lutz) [reply .generation/generate.py:212]: Otherwise, the generator would put "dist" there. At the moment the `dist` folder exists in the `main` branch. When ignoring that folder, it won't get updated.

File: python/.openapi-generator/FILES (hunk 3 above)
Comment (1lutz) [reply python/.openapi-generator/FILES:7]: This folder is entirely useless. I just deleted it: [8a54d46](https://github.com/geo-engine/openapi-client/pull/16/commits/8a54d462ae41d81eacd57909bd3fca19039f4e26)
